packages/core/src/components/MultiSelect/MultiSelect.test.tsx

```diff
@@ -194,6 +196,30 @@ describe('MultiSelect component', () => {
         expect(screen.queryByRole('list')).toBeVisible();
     });
 
+    it('should not show options if they do not match and hideOptionsOnNoResult is true', () => {
+        render(<MultiSelect options={options} hideOptionsOnNoResult={true} />);
+        fireEvent.change(screen.getByRole('textbox'), { target: { value: 'ValueThatDoesNotExist' } });
+        expect(screen.queryByText('Dummy2')).not.toBeInTheDocument();
+    });
+
+    it('should show all options if search query does not match and hideOptionsOnNoResult is false', () => {
+        render(<MultiSelect options={options} hideOptionsOnNoResult={false} />);
+        fireEvent.change(screen.getByRole('textbox'), { target: { value: 'ValueThatDoesNotExist' } });
+        expect(screen.getByText('Dummy2')).toBeInTheDocument();
+    });
+
+    it('should show no result component if no results are found and hideOptionsOnNoResult is true', () => {
```

Comment:
I think if `noResultComponent` is given then we don't need to check `hideOptionsOnNoResult`

Reply:
That makes sense. Thanks! Resolved.